### DISCUSSION TITLE
support textAlignment ALIGN_OPPOSITE on VerticalAxis labels

### DISCUSSION THREAD

**Discussion by jweick** [patrykandpatrick/vico#778] [Ideas]
Hi, 

i'm moving the bug my colleague opened (https://github.com/patrykandpatrick/vico/issues/692) to a feature-request, as you suggested.

I would suggest that the library supports setting the `textAlignment` on the `VerticalAxis`'s label to `ALIGN_OPPOSITE`, to get a smooth alignment, even when not using monospaced fonts.

As an image describes more than thousand words:
![graph](https://github.com/patrykandpatrick/vico/assets/932549/1a64c60b-4d72-4b1e-8349-cfcc03cb115d)

Keep up the great work!
cheers,
Jerg

**Comment by ahaglund** [comment]
I have a similar case where I'm not able to use the workaround explained in #692 because the values are of varying length. Would love to see this feature. 
Expected:
<img width="205" alt="image" src="https://github.com/patrykandpatrick/vico/assets/43466167/b133bdb7-26bb-46c5-b611-d403f4793555">
Actual:
<img width="383" alt="image" src="https://github.com/patrykandpatrick/vico/assets/43466167/225ccb87-6f88-4401-94d3-fdf0e2ab9979">

Thank you for your work!
Andreas

**Reply by Gowsky** [reply]
Hi @ahaglund, you could still apply the workaround by conditionally adding different numbers of leading spaces. But it would be easier to create a subclass, as @patrickmichalik mentioned below.

**Comment by patrickmichalik** [comment]
Hello, and thanks for the kind words! We’ll look into adding such functionality. In the meantime, you can achieve the desired result by creating a `VerticalAxis` subclass and making the required overrides, which are relatively minor. If you have any questions in this regard, let me know.

As per [this comment](https://github.com/patrykandpatrick/vico/issues/692#issuecomment-2183366633), if using a monospace typeface, you may also be able to implement this via a `CartesianValueFormatter` that adds leading spaces. This, however, is less robust and requires that the longest label’s character count be known at the formatting stage, which can prove harder to achieve than the aforementioned `VerticalAxis` subclass is to create. Thus, even with monospace typefaces, the subclass solution is generally preferable.
